### PR TITLE
LLT-5809

### PR DESCRIPTION
**Pull request by jjanowsk**
Some cleanup of UAPI handler. Details in commit messages.

### CODE REVIEW

File: neptun/src/device/integration_tests/mod.rs [hunk 1]
```diff
@@ -477,6 +473,33 @@ mod tests {
         assert!(response.ends_with("errno=0\n\n"));
     }
 
+    #[test]
+    #[ignore]
```
Comment (tomasz-grz) [inline neptun/src/device/integration_tests/mod.rs:477]: Is it right that we don't want to normally run these tests?

File: neptun/src/device/integration_tests/mod.rs [hunk 2]
```diff
@@ -477,6 +473,33 @@ mod tests {
...
+        let response = wg.wg_uapi_cmd("get=1\n\nget=1\n\n");
+        assert_eq!(response.matches("errno=0\n\n").count(), 2);
+
+        let response = wg.wg_uapi_cmd("get=1\n\nset=1\nlisten_port=12345\n\nget=1\n\n");
+        assert_eq!(response.matches("errno=0\n\n").count(), 3);
+    }
+
+    #[test]
+    #[ignore]
+    fn test_wireguard_uapi_set_without_proper_ending() {
```
Comment (tomasz-grz) [inline neptun/src/device/integration_tests/mod.rs:490]: What do you think about some tests that respond with errors?

File: neptun/src/device/api.rs [hunk 3]
```diff
@@ -148,26 +114,31 @@ pub fn api_exec<R: Read, W: Write>(
...
-        cmd.pop(); // pop the new line character
-        let status = match d.closed {
+    while status == 0 && reader.read_line(&mut cmd).is_ok_and(|n| n > 0) {
+        status = match d.closed {
             true => ENOENT,
             false => match cmd.as_ref() {
                 // Only two commands are legal according to the protocol, get=1 and set=1.
-                "get=1" => api_get(writer, d),
-                "set=1" => api_set(reader, d),
+                "get=1\n" => api_get(reader, writer, d),
```
Comment (tomasz-grz) [inline neptun/src/device/api.rs:124]: Do we really want to deal with all these new lines manually? What do you think about using something like `cmd.trim()` to sanitize the input, and `cmd.split_whitespace()` if there are multiple commands?

**Review by tomasz-grz — COMMENTED** [review]
Nice work, some questions since I'm not too familiar with the protocol.

**Review by dfetti — COMMENTED** [review]
+1.0 I'm not familiar with the project, but looks good to me. 
Might be a good idea to start a changelog file for this project?

File: neptun/src/device/integration_tests/mod.rs (hunk 1 above)
Comment (jjanowsk) [reply neptun/src/device/integration_tests/mod.rs:477]: That's just the way Boringtun used to differentiate integration tests. They are marked as ignored and you run them via `cargo tests --ignored`

File: neptun/src/device/api.rs (hunk 3 above)
Comment (jjanowsk) [reply neptun/src/device/api.rs:124]: Yes, we do. It's not about striping them. Stripping was (incorrectly) done earlier via `pop`. Here the point is that the \n is required. So sending "get=1[EOF]" is considered an incorrect command. It's just to keep compatibility with wireguard-go, nothing more.

File: neptun/src/device/integration_tests/mod.rs (hunk 2 above)
Comment (jjanowsk) [reply neptun/src/device/integration_tests/mod.rs:490]: Could add a few, but TBH I don't think there would be much value out of them.

**Review by tomasz-grz — APPROVED** [review]
+1 nice work

**Pull request merged**